### ISSUE TITLE
FEAT: Freeze Primary Key Column for Improved Usability in Large Tables

### ISSUE THREAD

**Issue by linaspasv** [beekeeper-studio/beekeeper-studio#2322]
Implement a feature that allows the primary key column to be frozen by default on large tables. This means that when the user scrolls to the right, the primary key column remains in place, making it easier to associate data with its respective ID.

**Benefits:**
- Users can easily track data across wide tables without losing sight of the primary key.
- Reduces the cognitive load on users, allowing for quicker data analysis and entry.

**Comment by Heinrich-ionline** [comment]
A better implementation of this feature would be to allow the user to select columns they would like to keep displaying when scrolling horizontally

**Comment by linaspasv** [comment]
> A better implementation of this feature would be to allow the user to select columns they would like to keep displaying when scrolling horizontally

This could be a nice extension to my suggestion, e.g. by default it could be a primary key but user could pick another column instead.

**Comment by not-night-but** [comment]
Yeah I think this could definitely be an improvement to the app. Pinning columns to the left, as well as maybe reordering columns in table view? This could be easily done in the same modal that we use to hide columns.
